test/controllers/tasks_controller_test.rb

```diff
@@ -79,48 +72,172 @@
 
       new_task = Task.find_by(name: task_hash[:task][:name])
       expect(new_task.description).must_equal task_hash[:task][:description]
-      expect(new_task.due_date.to_time.to_i).must_equal task_hash[:task][:due_date].to_i
-      expect(new_task.completed).must_equal task_hash[:task][:completed]
+      expect(new_task.completion_date).must_equal task_hash[:task][:completion_date]
 
       must_respond_with :redirect
       must_redirect_to task_path(new_task.id)
     end
   end
-
+  
   # Unskip and complete these tests for Wave 3
   describe "edit" do
     it "can get the edit page for an existing task" do
-      skip
-      # Your code here
+      get edit_task_path(task.id)
+      
+      must_respond_with :success
     end
-
+    
     it "will respond with redirect when attempting to edit a nonexistant task" do
-      skip
-      # Your code here
+      get edit_task_path(-1)
+      
+      must_respond_with :redirect
     end
   end
-
+  
   # Uncomment and complete these tests for Wave 3
   describe "update" do
-    # Note:  If there was a way to fail to save the changes to a task, that would be a great
-    #        thing to test.
     it "can update an existing task" do
-      # Your code here
+      updated_task_data = {
+        task: {
+          name: "Walk dog",
+          description: "Walk the dog",
+          completion_date: nil,
+        },
+      }
+      task_url = "/tasks/" + "#{task.id}"
+      
+      patch task_url, params: updated_task_data
```

Comment:
Instead of task_url, hopefully in the future I'd like to see this use the named path (in this case, it'd be `task_path`)